modules/300-prometheus/templates/prometheus/prometheus.yaml

```diff
@@ -144,6 +144,9 @@ spec:
       threshold.extended-monitoring.flant.com/disk-bytes-warning: "97"
       threshold.extended-monitoring.flant.com/disk-bytes-critical: "99"
       checksum/scraper: {{ include (print $.Template.BasePath "/prometheus-scraper-tls.yaml") . | sha256sum }}
+{{- if hasKey .Values.prometheus.internal.prometheusScraperIstioMTLS "certificate" }}
+      checksum/scraper-istio-mtls: {{ include (print $.Template.BasePath "/prometheus-scraper-istio-mtls.yaml") . | sha256sum }}
```

Comment:
Does it mean that Prometheus will be restarted every month?

Reply:
It does.

Reply:
Prometheus can hot reload certificates. The config reloader sidecar detects changes and triggers reloading.

The only problem is if a tcp session lasts for more than a client certificate lifetime,
but I thought the whole point to rotate certificate eleven month ahead was in avoiding excessive Prometheus restarts.

Reply:
I have an idea to remove the annotation with mtls secret checksum from the Prometheus resource and add an annotation checksum for PodMonitor and ServiceMonitor, so when the certificate changes, only these two resources will be recreated, and Prometheus will not restart.

Reply:
Let's check the Prometheus behaviour. @fl64 pls, change the mtls cert lifetime to 5min and check if prometheus watch the cert.

Reply:
@apolovov 
The test was completed successfully. Prometheus successfully reloads certificates if they are changed.